lesson03/assignment/index.js

```diff
@@ -0,0 +1,144 @@
+    ]
+  },
+  {
+    "name": "Barbara Bonansea",
+    "jobTitle": "Software Engineer",
+    "company": "Excellence in the Cloud",
+    "experience": "12 years",
+    "school": "University of Southern California",
+    "major": "Computer Science",
+    "email": "barbara@example.com",
+    "linkedInUrl": "barbara.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "JavaScript", "C", "Go"
+    ]
+  },
+  {
+    "name": "Javier Hernandez",
+    "jobTitle": "User Experience Engineer",
+    "company": "Web Sites and More",
+    "experience": "5 years",
+    "school": "Seattle University",
+    "major": "Performing Arts",
+    "email": "javier@example.com",
+    "linkedInUrl": "javier.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "CSS"
+    ]
+  },
+  {
+    "name": "Maribel Dominguez",
+    "jobTitle": "Front End Engineer",
+    "company": "Bits and Bytes",
+    "experience": "6 years",
+    "school": "University of Washington",
+    "major": "Mechanical Engineering",
+    "email": "maribel@example.com",
+    "linkedInUrl": "maribel.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "CSS", "JavaScript", "React", "Vue", "Redux"
+    ]
+  }
+]
+
+// select DOM node
+const userProfileContainer = document.querySelector(".template-hook");
+
+// Update userProfiles entry counter
+let i =+ 0;
+
+//============forEach DOM NODE FUNCTION============
+//run a forEach on userProfiles array
+userProfiles.forEach(function(el) { //passes each property name/value pair from userProfiles' array of objects to the el paramenter as arguement to function below.
+  //Create new HTML section
+  let nametag = document.createElement('section')
+  nametag.innerHTML = `
+  <h1>${el.name}</h1>
+  `
+
+  //==========Populate nametag section with object properties values from userProfiles.=========
+  //For below, can you automatically ouput property names too?
+  let card__details = document.createElement('ul')
+  card__details.innerHTML = `
+  <li><h3>Job Title: </h3><h4>${el.jobTitle}</h4></li>
+  <li><h3>Company: </h3><h4>${el.company}</h4></li>
+  <li><h3>Experience: </h3><h4>${el.experience}</h4></li>
+  <li><h3>Education: </h3><h4>${el.school}</h4></li>
+  <li><h3>Degree: </h3><h4>${el.major}</h4></li>
+  <li><h3>Email: </h3><h4>${el.email}</h4></li>
+  <li><h3>LinkedIn: </h3><h4>${el.linkedInUrl}</h4></li>
+  <li><h3>Code Proficiencies: </h3></li>
+  `
+
+  //============Code Proficiencies by Entry. DOM output.============
+  //create nested ul element for code languages
+  let codeEntry = document.createElement('ul')
+
+  //reset x counter (codeLanguages array length)
+  let x = 0;
+  for (x; userProfiles[i].codeLanguages.length > x; x++) { //loop runs for array codeLanguages.length number of times
+    let newCodeLang = document.createElement('li')
+    newCodeLang.innerHTML = `
+    <li><h4>${userProfiles[i].codeLanguages[x]} </h4></li>
+    `
+      card__details.appendChild(codeEntry)
```

Comment:
This line should be outside the loop, down on line 108. For each codeLanguage, you want to create that `li` and append it to the `ul`, but the `ul` only needs to be appended to userProfileContainer once, after all looping has completed.